prime-router/src/main/kotlin/azure/EmailEngineFunction.kt

```diff
@@ -304,26 +304,32 @@ class EmailScheduleEngine {
         var emails: MutableList<String> = mutableListOf()
 
         try {
-            var ssws: String? = SecretHelper.getSecretService().fetchSecret("SSWS_OKTA")
+            val ssws: String? = SecretHelper.getSecretService().fetchSecret("SSWS_OKTA")
 
             if (ssws !== null) {
-                var grp = encodeOrg(org)
+                val grp = encodeOrg(org)
 
                 // get the OKTA Group Id
-                var (_, _, response1) = Fuel.get("$OKTA_GROUPS_API?q=$grp")
-                    .header(mapOf("Authorization" to "SSWS $ssws")).responseJson()
-                var grpId = ((response1.get().array()).get(0) as JSONObject).getString("id")
+                val (_, respStrJson1) = HttpClientUtils.getWithStringResponse(
+                    url = "$OKTA_GROUPS_API?q=$grp",
+                    headers = mapOf("Authorization" to "SSWS $ssws")
+                )
+
+                val grpId = JSONObject(respStrJson1).getString("id")
 
                 // get the users within that OKTA group
-                var (_, _, response) = Fuel.get("$OKTA_GROUPS_API/$grpId/users")
-                    .header(mapOf("Authorization" to "SSWS $ssws")).responseJson()
+                val (_, respStrJson) = HttpClientUtils.getWithStringResponse(
+                    url = "$OKTA_GROUPS_API/$grpId/users",
+                    headers = mapOf("Authorization" to "SSWS $ssws")
+                )
 
-                for (user in response.get().array()) emails.add(
+                val users = JSONArray(respStrJson)
+                for (user in users) emails.add(
                     (user as JSONObject).getJSONObject("profile").getString("email")
                 )
             }
         } catch (ex: Throwable) {
-            logger.warning("Error in fetching emails")
+            logger.warning("Error in fetching emails, exception: ${ex.message}")
```

Comment:
move this to 

```kotlin
logger.warning("Error in fetching emails, exception", ex)
```

So it triggers errors and we catch that something is amiss.

Reply:
I wasn't able to chain that exception to `logger.warn`, maybe I'm missing something?